Userland/DevTools/UserspaceEmulator/SoftFPU.h

```diff
@@ -0,0 +1,612 @@
+/*
+ * Copyright (c) 2021, Leon Albrecht <leon2002.la@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "Report.h"
+#include <AK/Concepts.h>
+#include <AK/SIMD.h>
+#include <LibX86/Instruction.h>
+#include <LibX86/Interpreter.h>
+
+#include <math.h>
+#include <string.h>
+
+namespace UserspaceEmulator {
+using namespace AK::SIMD;
+class Emulator;
+class SoftCPU;
+
+union MMX {
```

Comment:
This is going to be hard on ARM in BE mode to get right... Don't know if we care though.

Reply:
I think the whole UE isn't very port friendly (the dependency is called LibX86 after all)
according to what I know of arm, there are 3 floating point register sets in addition to simd ones, but the state manegement would be way simpler so it could fit into the main cpu emulation without harming readability that much

Reply:
Yeah the purpose of UE is to run the current userland in a virtualized environment for debugging. Running a cross-compiled userland is not really a goal :)

Reply:
ARM FP is so different in implementation (especially NEON) that I don't think the term "register sets" even applies to it. 

It is quite entangled with the x86 cpu, yes. Maybe extract the x86/87 state into a struct & give both CPU and FPU access to that?